File: python-client-sdk/src/orbapi/models/operations/post_link_exchange.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from orbapi import utils
from ..shared import security as shared_security


@dataclass_json
@dataclasses.dataclass
class PostLinkExchangeRequestBody:
    public_token: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('publicToken') }})
    

@dataclasses.dataclass
class PostLinkExchangeSecurity:
    vessel_api_token: shared_security.SchemeVesselAPIToken = dataclasses.field(metadata={'security': { 'scheme': True, 'type': 'apiKey', 'sub_type': 'header' }})
    
class PostLinkExchangeResponseBodyIntegrationIDEnum(str, Enum):
    SALESFORCE = "salesforce"
    HUBSPOT = "hubspot"
    PIPEDRIVE = "pipedrive"


@dataclass_json
@dataclasses.dataclass
class PostLinkExchangeResponseBody:
    access_token: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('accessToken') }})
    connection_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('connectionId') }})
    integration_id: PostLinkExchangeResponseBodyIntegrationIDEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('integrationId') }})
    native_org_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('nativeOrgId') }})
    native_org_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('nativeOrgURL') }})
    

@dataclasses.dataclass
class PostLinkExchangeRequest:
    security: PostLinkExchangeSecurity = dataclasses.field()
    request: Optional[PostLinkExchangeRequestBody] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    

@dataclasses.dataclass
class PostLinkExchangeResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    response_body: Optional[PostLinkExchangeResponseBody] = dataclasses.field(default=None)
    

File: typescript-client-sdk/src/sdk/models/operations/postlinkexchange.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PostLinkExchangeRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=publicToken" })
  publicToken: string;
}


export class PostLinkExchangeSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  vesselAPIToken: shared.SchemeVesselApiToken;
}

export enum PostLinkExchangeResponseBodyIntegrationIdEnum {
    Salesforce = "salesforce",
    Hubspot = "hubspot",
    Pipedrive = "pipedrive"
}


export class PostLinkExchangeResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "json, name=connectionId" })
  connectionId: string;

  @SpeakeasyMetadata({ data: "json, name=integrationId" })
  integrationId: PostLinkExchangeResponseBodyIntegrationIdEnum;

  @SpeakeasyMetadata({ data: "json, name=nativeOrgId" })
  nativeOrgId: string;

  @SpeakeasyMetadata({ data: "json, name=nativeOrgURL" })
  nativeOrgURL: string;
}


export class PostLinkExchangeRequest extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PostLinkExchangeRequestBody;

  @SpeakeasyMetadata()
  security: PostLinkExchangeSecurity;
}


export class PostLinkExchangeResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: PostLinkExchangeResponseBody;
}
